src/cdk/auto-discover-base.ts

```diff
@@ -0,0 +1,70 @@
+import { join } from "path";
+import * as glob from "glob";
+import { Component } from "../component";
+import { Project } from "../project";
+import { TYPESCRIPT_INTEG_EXT } from "./internal";
+
+/**
+ * Options for `AutoDiscoverBase`
+ */
+export interface AutoDiscoverBaseOptions {
+  /**
+   * Locate files with the given extension.
+   *
+   * @example ".integ.ts"
+   */
+  readonly extension: string;
+
+  /**
+   * Locate entrypoints in the given project directory.
+   *
+   * @example "test"
+   */
+  readonly projectdir: string;
+}
+
+/**
+ * Base class for auto-discovering and creating project subcomponents.
+ */
+export class AutoDiscoverBase extends Component {
```

Comment:
Let's make `AutoDiscoverBase` and `IntegrationTestBase` abstract for now, since they don't do much on their own (minimizing API surface area):
```suggestion
export abstract class AutoDiscoverBase extends Component {
```
This might require updating tests to use a stub child class